src/security/builtin_plugins/authentication/src/authentication.c

```diff
@@ -269,17 +271,28 @@ security_object_find(
     return (SecurityObject *) ddsrt_hh_lookup(hh, &template);;
 }
 
+static int compare_relation (const void *va, const void *vb)
+{
+    const struct RemoteIdentityInfo * const *ha = va;
+    const struct RemoteIdentityInfo * const *hb = vb;
+
+    if (*ha < *hb) {
```

Comment:
I think this should be comparing ``*ha`` and ``*hb`` cast to ``uintptr_t``, though it is possible I am mistaken in thinking that pointer comparison is among the operations causing undefined behaviour when the addresses come from unrelated objects.